msgraph/client.go

```diff
@@ -161,100 +149,73 @@ func (c Client) performRequest(req *http.Request, input HttpRequestInput) (*http
 		}
 	}
 
-	var attempts, backoff, multiplier int64
-	for attempts = 0; attempts < requestAttemptsForRateLimiting; attempts++ {
-		// sleep after the previous failed attempt
-		if attempts > 0 {
-			time.Sleep(time.Duration(backoff))
-		}
+	c.retryableClient.CheckRetry = func(ctx context.Context, resp *http.Response, err error) (bool, error) {
```

Comment:
because this function sits on the instance of the object `(Client)` and not `(*Client`) this change to retryableClient won't be retained, which seems fine but just double-checking

Reply:
This was the intent actually, as the `ConsistencyFailureFunc` is defined per operation / request type, e.g. a consistency error looks different for say `GET /foo/1` to `PUT /foo/1`